graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsSchemaProvider.kt

```diff
@@ -426,7 +431,16 @@ class DgsSchemaProvider(
             }
         }
 
-        return ReflectionUtils.invokeMethod(method, dgsComponent, *args.toTypedArray())
+        return if (method.kotlinFunction?.isSuspend == true) {
```

Comment:
I think there's some helpers from spring-core we could use;

Here's one for checking if a method is suspending:

https://github.com/spring-projects/spring-framework/blob/b595dc1dfad9db534ca7b9e8f46bb9926b88ab5a/spring-core/src/main/java/org/springframework/core/KotlinDetector.java#L82

Here is one for invoking a suspending function:

https://github.com/spring-projects/spring-framework/blob/3dba3691a7cee4299241e1f12d1680788ea77731/spring-core/kotlin-coroutines/src/main/kotlin/org/springframework/core/CoroutinesUtils.kt#L60

Reply:
It turns out the helper methods are not available yet in the Spring version we're depending on - the methods are all "internal".
It makes sense to use this when we bump the Spring version though.

Reply:
CoroutinesUtils.invokeSuspendingFunction is public, and was added in Spring 5.2, which is what we depend on. You're right about the other one, isSuspendingFunction was added in 5.3 unfortunately.

Reply:
Nevermind, I see even CoroutinesUtils.invokeSuspendingFunction was internal until 5.3...